meshtastic/interfaces/ble/lifecycle_service.py

```
@@ -0,0 +1,797 @@
+            iface.auto_reconnect, iface._shutdown_event
+        )
+
+    @staticmethod
+    def _disconnect_and_close_client(
+        iface: "BLEInterface", client: "BLEClient"
+    ) -> None:
+        """Ensure BLE client resources are released."""
+        iface._client_manager_safe_close_client(client)
+
+    @staticmethod
+    def _handle_disconnect(
```

Ruff (0.15.5)

[warning] Too many return statements (9 > 6)

(PLR0911)